### PR TITLE
Development

### PR DESCRIPTION
**Pull request by wskopalik**
As discussed in today's meeting, please review the uploaded upLED and upTEMP sample code for Python.

Thanks,

Wolfgang

### CODE REVIEW

File: Python/upLED and upTEMP Controllers/upLED and upTEMP - Python 3 using ctypes.py [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+    currentSetpoint = c_double()
+    LEDName = create_string_buffer(256)
+    LEDSerialNumber = create_string_buffer(256)
+    LEDCurrentLimit = c_double()
+    LEDForwardVoltage = c_double()
+    LEDWavelength = c_double()
+    
+    lib.TLUP_getLedInfo(upHandle, LEDName, LEDSerialNumber, byref(LEDCurrentLimit),
+                        byref(LEDForwardVoltage), byref(LEDWavelength))
+    print("Connected LED:", (LEDName.value).decode(), ", serial number:", (LEDSerialNumber.value).decode(),", wavelength [nm]:",LEDWavelength.value)
```
Comment (manzelmi-thorlabs) [inline Python/upLED and upTEMP Controllers/upLED and upTEMP - Python 3 using ctypes.py]: We should add this to an IF statement that checks if the return from the LEDInfo command is null so the print will be skipped. If null, set the current limit to something else so the code doesn't error.

File: Python/upLED and upTEMP Controllers/upLED and upTEMP - Python 3 using ctypes.py [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+    print()
+
+    #Set the current setpoint in Ampere to 10% of the current limit of this LED.
+    currentSetpoint = c_double((0.1) * LEDCurrentLimit.value)
+    lib.TLUP_setLedCurrentSetpoint(upHandle,currentSetpoint)
+    print("LED current set to [A]:", currentSetpoint.value)
+    print()
+
+    #Switch the LED on, wait for 2 sec, then switch it off again.
+    lib.TLUP_switchLedOutput(upHandle,1)
```
Comment (manzelmi-thorlabs) [inline Python/upLED and upTEMP Controllers/upLED and upTEMP - Python 3 using ctypes.py:93]: Is there a chance of these methods throwing errors? Can we show how to handle them?

File: Python/upLED and upTEMP Controllers/upLED and upTEMP - Python 3 using ctypes.py [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+    steinhartHartParameterE = c_double()
+    steinhartHartParameterF = c_double()    
+
+    lib.TLUP_getSteinhartHartParameterFromExponentialParameterConvertion(
+        upHandle, t0, r0, beta,
+        byref(steinhartHartParameterA), byref(steinhartHartParameterB),
+        byref(steinhartHartParameterC), byref(steinhartHartParameterD),
+        byref(steinhartHartParameterE), byref(steinhartHartParameterF))
+
+    """#Get possible temperature ranges for these Steinhart-Hart-Parameters.
```
Comment (manzelmi-thorlabs) [inline Python/upLED and upTEMP Controllers/upLED and upTEMP - Python 3 using ctypes.py]: Should this be uncommented?

File: Python/upLED and upTEMP Controllers/upLED and upTEMP - Python 3 using ctypes.py [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+    print()"""
+
+    #Channel 1 is configured with the calculated Steinhart-Hart coefficients.
+    CH1 = c_int(1)
+    lib.TLUP_setThermSensConfig(upHandle, CH1, c_int(3),
+                                steinhartHartParameterA, steinhartHartParameterB,
+                                steinhartHartParameterC, steinhartHartParameterD,
+                                steinhartHartParameterE, steinhartHartParameterF)
+
+    """#Channel list is created and enabled.
```
Comment (manzelmi-thorlabs) [inline Python/upLED and upTEMP Controllers/upLED and upTEMP - Python 3 using ctypes.py]: Is this to measure all channels at once? Comment might need to be longer for this if it is needed

**Review by manzelmi-thorlabs — COMMENTED** [review]
Please see comments here. Code looks good but some minor changes are needed. Thanks!

**Review by TrumanMetz — APPROVED** [review]

**Pull request closed without merging**